core/.js/src/main/scala/io/udash/routing/UrlChangeProvider.scala

```diff
@@ -132,5 +127,5 @@ object WindowUrlPathChangeProvider extends UrlChangeProvider {
   }
 
   override def currentFragment: Url =
-    Url(Option(window.history.state).map(_.asInstanceOf[js.Dynamic].url.asInstanceOf[String]).getOrElse(window.location.pathname))
+    Url(window.history.state.opt.map(_.asInstanceOf[js.Dynamic].url.toString).getOrElse(window.location.pathname))
```

Comment:
Could `Url` be serialized directly as state data? So that you wouldn't have to recreate it each time you query for current fragment?

Reply:
It shouldn't be created - it's a value class. I'm actually using the URL part of the state itself here, not the state data itself (which seems to be a redundant copy of the same thing). I'll try to remove it from the state data altogether to avoid confusion.